leapp/utils/report.py

```diff
@@ -139,4 +139,13 @@ def generate_report_file(messages_to_report, context, path, report_schema='1.1.0
                     groups = msg.pop('groups', [])
                     msg['flags'] = [g for g in groups if g in _DEPRECATION_FLAGS]
                     msg['tags'] = [g for g in groups if g not in _DEPRECATION_FLAGS]
+            if report_schema_tuple == (1, 2, 0):
+                # DEPRECATED: flags, tags
+                # NOTE(pstodulk): This is a temporary solution to ensure that
+                # flags are not present in the JSON output as we need to re-introduce
+                # flags internally to ensure that any inhibitor checks will not
+                # be negatively affected until we drop flags and tags completely.
+                messages_to_report = list(messages_to_report)
+                for msg in messages_to_report:
+                    msg.pop('flags')
```

Comment:
Your patch creates `flags` field only in case report has an inhibitor, but here you are popping `flags` from any message. I suspected this to crash and it really does in case of report-schema 1.2.0
```
Traceback (most recent call last):

  File "/usr/bin/leapp", line 11, in <module>

    load_entry_point('leapp==0.14.0', 'console_scripts', 'leapp')()

  File "/usr/lib/python3.6/site-packages/leapp/cli/__init__.py", line 43, in main

    cli.command.execute('leapp version {}'.format(VERSION))

  File "/usr/lib/python3.6/site-packages/leapp/utils/clicmd.py", line 106, in execute

    args.func(args)

  File "/usr/lib/python3.6/site-packages/leapp/utils/clicmd.py", line 128, in called

    self.target(args)

  File "/usr/lib/python3.6/site-packages/leapp/cli/commands/upgrade/breadcrumbs.py", line 94, in wrapper

    return f(*args, breadcrumbs=breadcrumbs, **kwargs)

  File "/usr/lib/python3.6/site-packages/leapp/cli/commands/preupgrade/__init__.py", line 73, in preupgrade

    util.generate_report_files(context, report_schema)

  File "/usr/lib/python3.6/site-packages/leapp/cli/commands/upgrade/util.py", line 147, in generate_report_files

    generate_report_file(messages, context, report_json, report_schema)

  File "/usr/lib/python3.6/site-packages/leapp/utils/report.py", line 150, in generate_report_file

    msg.pop('flags')

KeyError: 'flags'
```
http://artifacts.osci.redhat.com/testing-farm/255b2826-8ee9-471f-881e-30b94fa85fef/

Reply:
`dict.pop` and `dict.get` are annoying, pop behaves differently if no default parameter was passed.

Reply:
nice catch.